registry/infrastructure/repositories/organization_repository.py

```diff
@@ -58,13 +61,35 @@ def update_org_draft(self, current_draft, organization, username):
         current_draft.Organization.description = organization.description
         current_draft.Organization.short_description = organization.short_description
         current_draft.Organization.url = organization.url
+        current_draft.Organization.duns_no = organization.duns_no
         current_draft.Organization.contacts = organization.contacts
         current_draft.Organization.assets = organization.assets
         current_draft.Organization.metadata_ipfs_hash = organization.metadata_ipfs_hash
         current_draft.OrganizationReviewWorkflow.updated_by = username
         current_draft.OrganizationReviewWorkflow.updated_on = datetime.utcnow()
+        self.session.query(OrganizationAddress).filter(
+            OrganizationAddress.org_row_id == current_draft.Organization.row_id).delete(synchronize_session='fetch')
```

Comment:
please move this to another function , and make it explicit that we are deleting all old record and adding new ones